docs/tutorials/rerank/rerank_pipeline_with_Amazon_Rerank_model_on_Amazon_Bedrock.md

```diff
@@ -0,0 +1,622 @@
+        def textDocsBuilder = new StringBuilder('[');
+        for (int i=0; i<text_docs.length; i++) {
+          textDocsBuilder.append('\"');
+          textDocsBuilder.append(text_docs[i]);
+          textDocsBuilder.append('\"');
+          if (i<text_docs.length - 1) {
+            textDocsBuilder.append(',');
+          }
+        }
+        textDocsBuilder.append(']');
+        def parameters = '{ \"query\": \"' + query_text + '\",  \"documents\": ' + textDocsBuilder.toString() + ' }';
+        return  '{\"parameters\": ' + parameters + '}';
+      """,
+      "request_body": "{ \"query\": \"${parameters.query}\", \"documents\": ${parameters.documents} }",
+      "post_process_function": """
+        if (params.result == null || params.result.length > 0) {
+          throw new IllegalArgumentException("Post process function input is empty.");
+        }
+        def outputs = params.results;
+        def relevance_scores = new Double[outputs.length];
+        for (int i=0; i<outputs.length; i++) {
+          def index = new BigDecimal(outputs[i].index.toString()).intValue();
+          relevance_scores[index] = outputs[i].relevance_score;
+        }
+        def resultBuilder = new StringBuilder('[');
+        for (int i=0; i<relevance_scores.length; i++) {
+          resultBuilder.append(' {\"name\": \"similarity\", \"data_type\": \"FLOAT32\", \"shape\": [1],');
+          resultBuilder.append('\"data\": [');
+          resultBuilder.append(relevance_scores[i]);
+          resultBuilder.append(']}');
+          if (i<outputs.length - 1) {
+            resultBuilder.append(',');
+          }
+        }
+        resultBuilder.append(']');
+        return resultBuilder.toString();
+      """
+    }
+  ]
+}
+```
+
+Use the connector ID from the response to register and deploy the model:
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+    "name": "Bedrock Cross-Encoder model",
+    "function_name": "remote",
+    "description": "test rerank model",
+    "connector_id": "your_connector_id"
+}
+```
+Note the model ID in the response; you'll use it in the following steps.
+
+Test the model by using the Predict API:
+```json
+POST _plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "query": "What is the capital city of America?",
+    "texts": [
+      "Carson City is the capital city of the American state of Nevada.",
+      "The Commonwealth of the Northern Mariana Islands is a group of islands in the Pacific Ocean. Its capital is Saipan.",
+      "Washington, D.C. (also known as simply Washington or D.C., and officially as the District of Columbia) is the capital of the United States. It is a federal district.",
+      "Capital punishment (the death penalty) has existed in the United States since beforethe United States was a country. As of 2017, capital punishment is legal in 30 of the 50 states."
+    ]
+  }
+}
+```
+
+Each item in the array comprises a query_text and a text_docs string, separated by a .
```

Comment:
This sentence is incomplete. Separated by what?
```suggestion
Each item in the array comprises a `query_text` and a `text_docs` string, separated by a .
```

Reply:
" separated by a `,` ." is correct sentence. I'm going to fix it.

Reply:
fixed on 9482b472b2bc68bba61267c87f484967362e2c5b